---
title: "MICB425 Portfolio - Microbial Ecological Genomics: From Genomes to Biomes, Spring 2018"
author: "Paaksum Wong, 18315144"
date: "version `r format(Sys.time(), '%B %d, %Y')`"
output: 
  html_document:
    toc: true
---
# Module 01
Reserve the first level headings (#) for the start of a new Module. This will help to organize your portfolio in an intuitive fashion.  

## Module 01 portfolio check
The first of your second level headers (##) is to be used for the portfolio content checks. The Module 01 portfolio check has been built for you directly into this template, but will also be available as a stand-alone markdown document available on the MICB425 GitHub so that you know what is required in each module section in your portfolio. The completion status and comments will be filled in by the instructors during portfolio checks when your current portfolios are pulled from GitHub.
    
* Evidence worksheet_01
    + Completion status: X
    + Comments: Check header. Does not appear in table of contents.
* Evidence worksheet_02
    + Completion status: X
    + Comments:
* Evidence worksheet_03
    + Completion status: X
    + Comments:
* Problem Set_01
    + Completion status: X
    + Comments:
* Problem Set_02
    + Completion status: X
    + Comments: Expand explanations. 1 question not answered.
* Writing assessment_01
    + Completion status: X
    + Comments:
* Additional Readings
    + Completion status: X
    + Comments: Needs links.
* Installation check
    + Completion status: X
    + Comments:
* Portfolio repo setup
    + Completion status: X
    + Comments:
* RMarkdown Pretty PDF Challenge
    + Completion status: X
    + Comments:
* ggplot
    + Completion status: 9/10
    + Comments: Make sure your "fill" aesthetics are the same for all layers of the plot. See how there are species and phyla names in the legend.

## Data science Friday  
The remaining second level headers (##) are for separating data science Friday, regular course, and project content. In this module, you will only need to include data science Friday and regular course content; projects will come later in the course.

### Installation check
Third level headers (###) should be used for links to assignments, evidence worksheets, problem sets, and readings, as seen here.  

Use this space to include your installation screenshots.

![Git Bash Terminal](Git Bash Terminal.jpg)
![RStudio Screenshot](RStudio Screenshot.jpg)
![GitHub Homepage](GitHub Homepage.jpg)

### Portfolio repo setup
Detail the code you used to create, initialize, and push your portfolio repo to GitHub. This will be helpful as you will need to repeat many of these steps to update your porfolio throughout the course.

Pushing files to GitHub:
Navigate to portfolio directory by using "cd ~/Documents" and "cd MICB425_portfolio".
"git.add"
"git commit -m "nth commit""
"git push"
Verify on GitHub it was successfully pushed.

### R Markdown Pretty html Challenge
Refer to 18315144_Assignment3 markdown file in portfolio.

### ggplot Exercises
```{r}
library(tidyverse)
source("https://bioconductor.org/biocLite.R")
biocLite("phyloseq")
library("phyloseq")
load("phyloseq_object.RData")
metadata=read.table(file="Saanich.metadata.txt", header=TRUE, row.names=1, sep="\t")
ggplot(metadata, aes(x=O2_uM, y=Depth_m))+
  geom_point(shape="square")
ggplot(metadata, aes(x=O2_uM, y=Depth_m))+
  geom_point(size=1)
ggplot(metadata, aes(x=O2_uM, y=Depth_m, size=OxygenSBE_V))+
  geom_point()

#Exercise 1
ggplot(metadata, aes(x=PO4_uM, y=Depth_m))+
  geom_point(shape=17, color="purple")

#Exercise 2
metadata %>% 
  mutate(Temperature_F=Temperature_C*1.8+32) %>% 
  ggplot(aes(x=Temperature_F, y=Depth_m))+
  geom_point()

#gglot with phyloseq
physeq
plot_bar(physeq, fill="Phylum")
physeq_percent=transform_sample_counts(physeq, function(x) 100 * x/sum(x))
plot_bar(physeq, fill = "Phylum")
plot_bar(physeq_percent, fill = "Phylum")+
  geom_bar(aes(fill=Phylum), stat="identity")

#Exercise 3
plot_bar(physeq_percent, fill = "Phylum")+
  geom_bar(aes(fill=Phylum), stat="identity")+
  labs(title="Phyla from 10 to 200 m in Saanich Inlet", x="Sample depth", y="Percent relative abundance")

#Faceting
plot_bar(physeq_percent, fill = "Phylum")+
  geom_bar(aes(fill=Phylum), stat="identity")+
  facet_wrap(~Phylum)
plot_bar(physeq_percent, fill = "Phylum")+
  geom_bar(aes(fill=Phylum), stat="identity")+
  facet_wrap(~Phylum, scales = "free_y")+
  theme(legend.position = "none")

#Exercise 4
metadata %>% 
  gather(nutrients, uM, O2_uM, PO4_uM, SiO2_uM, NO3_uM, NH4_uM, NO2_uM) %>% 
  ggplot(aes(Depth_m, uM))+
  geom_point()+
  geom_line()+
  facet_wrap(~nutrients, scales = "free_y")+
  theme(legend.position = "none")

```

## Origins and Earth Systems

### Evidence worksheet 01 

[Whitman *et al* 1998](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC33863/)

#### Learning objectives
Describe the numerical abundance of microbial life in relation to ecology and biogeochemistry of Earth systems. 

#### General questions
* What were the main questions being asked?  
What are the primary habitats of prokaryotes? What is their abundance and total carbon content in each of the main habitats? What is the total nutrient content in these habitats?

* What were the primary methodological approaches used?  
The primary methodological approaches included estimating cellular density, volume, and carbon to estimate the number of prokaryotes in different habitats. Extrapolation of cell densities were used to determine number of subsurface prokaryotes. To determine amount of carbon in prokaryotes, cellular carbon content was assumed to be half the dry weight.

* Summarize the main results or findings.  
The main habitats of prokaryotes and their number of prokaryotes are aquatic habitats (1.2x10^29^), oceanic subsurface (3.5x10^30^), soil (2.6x10^29^), and terrestrial subsurface (0.25-2.5x10^30^). The total number of prokaryotic cells on Earth is 4-6x10^30^ and their total carbon content is 350-550Pg. They also contain 85-130Pg N and 9-14Pg P. The cellular production rate of all prokaryotes on Earth is estimated to be 1.7x10^30^ cells/yr and is highest in the open ocean.

* Do new questions arise from the results?  
What role do these large number of prokaryotes play in maintaining the ecosystem of their various habitats? What is the significance of differences in  turnover rates for prokaryotic populations in different habitats? How can methods to estimate carbon content in prokaryotes be improved upon, especially in habitats where sampling is difficult (eg. deep subsurface)?

* Were there any specific challenges or advantages in understanding the paper (*e.g.* did the authors provide sufficient background information to understand experimental logic, were methods explained adequately, were any specific assumptions made, were conclusions justified based on the evidence, were the figures or tables useful and easy to understand)?  
The paper provided a useful insight regarding the abundance of prokaryotes and their carbon content on Earth. However, the validity of the methods and sources the authors used to estimate these numbers could be more robust in that many interpretations were estimates and extrapolations.


### Problem set 01

[Whitman *et al* 1998](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC33863/)

#### Learning objectives: 
Describe the numerical abundance of microbial life in relation to the ecology and biogeochemistry of Earth systems.

#### Specific questions:
* What are the primary prokaryotic habitats on Earth and how do they vary with respect to their capacity to support life? Provide a breakdown of total cell abundance for each primary habitat from the tables provided in the text.  
The primary prokaryotic habitats on Earth are aquatic environments (cell abundance=1.2x10^29^), soils (cell abundance=2.5x10^29^), and oceanic/terrestrial subsurface (cell abundance=3.8x10^30^). 

* What is the estimated prokaryotic cell abundance in the upper 200 m of the ocean and what fraction of this biomass is represented by marine cyanobacterium including Prochlorococcus? What is the significance of this ratio with respect to carbon cycling in the ocean and the atmospheric composition of the Earth?  
The estimated prokaryotic cell abundance in the upper 200m of the ocean is 3.6x10^28^, with a density of 5x10^5^ cells/ml. The cell density of cyanobacteria is 4x10^4^ cells/ml, so the ratio of cyanobacteria over total prokaryotic cells is 8%.

* What is the difference between an autotroph, heterotroph, and a lithotroph based on information provided in the text?  
An autotroph is "self-nourishing", they fix inorganic carbon (CO2). Heterotrophs assimilate organic carbon, and lithotrophs use inorganic substrates.

* Based on information provided in the text and your knowledge of geography what is the deepest habitat capable of supporting prokaryotic life? What is the primary limiting factor at this depth?  
The deepest habitat capable of supporting prokaryotic life is deep subsurface habitats (terrestrial is 4km and marine is ~14.9km (~4km below the ocean floor. Mariana's Trench is 10.9km below sea level)). The temperature is the limiting factor at this depth. The temperature change is roughly 22 degrees Celsius per km.

* Based on information provided in the text your knowledge of geography what is the highest habitat capable of supporting prokaryotic life? What is the primary limiting factor at this height?  
The highest habitat capable of supporting prokaryotic life is 8.8km at Mount Everest, the highest point on Earth. Although prokaryotes do exist in the stratosphere to some extent such as spore-forming microbes, there are no nutrients and moisture, and there is strong ionizing radiation. 

* Based on estimates of prokaryotic habitat limitation, what is the vertical distance of the Earth's biosphere measured in km?  
14.9km below sea level + 8.8km above sea level = 23.7km is the vertical distance of the Earth's bioshpere.

* How was annual cellular production of prokaryotes described in Table 7 column four determined? (Provide an example of the calculation)  
Population x turnover rate/yr = cells/yr  
3.6x10^28^ cells x 365 days / 16 turnovers = 8.2x10^29^ cells/yr

* What is the relationship between carbon content, carbon assimilation efficiency and turnover rates in the upper 200m of the ocean? Why does this vary with depth in the ocean and between terrestrial and marine habitats?  
Carbon efficiency = 20%, and 5-20fg C/cell (10^-30^pg/cell). There are 3.6x10^28^ cells.  
3.6x10^28^ cells x 20x10^-30^ Pg/cell = 0.72Pg C/marine heterotrophs  
0.72x4 = 2.88Pg/yr  
51Pg C/yr x 85% consumed = 43Pg C  
43Pg C/yr / 2.88Pg/yr = 14.9, 1 turnover every 24.5 days.  
*Numbers were pulled from multiple sources of literature, may not be an accurate calculation*
 
* How were the frequency numbers for four simultaneous mutations in shared genes determined for marine heterotrophs and marine autotrophs given an average mutation rate of 4 x 10^-7^ per DNA replication? (Provide an example of the calculation with units. Hint: cell and generation cancel out)  
(4x10^-7^)^4^=2.56x10^-26^ mutations/generation  
3.1x10^28^ cells x (365/16 turnovers/yr) = 8.2x10^29^ cells/yr  
8.2x10^29^ cells/yr x 2.56x10^-26^ mutations/generation = 2.1x10^4^ mutations/yr

* Given the large population size and high mutation rate of prokaryotic cells, what are the implications with respect to genetic diversity and adaptive potential? Are point mutations the only way in which microbial genomes diversify and adapt?  
With the large population size and high mutation rate, prokaryotic cells have a very high degree of genetic diversity and adaptive potential, as traits that were selected for from the diverse pool were proliferated quickly and in high numbers. In addition to point mutations, microbial genomes also diversify and adapt by lateral gene transfer.

* What relationships can be inferred between prokaryotic abundance, diversity, and metabolic potential based on the information provided in the text?  
The abundance and high diversity of prokaryotes coupled to their ability to rapidly proliferate means they have a high metabolic potential.

### Evidence worksheet 02 

[E. G. Nisbet & N. H. Sleep 2001](https://www.ncbi.nlm.nih.gov/pubmed/?term=The+habitat+and+nature+of+early+life+nisbet)  
[James F. Kasting and Janet L. Siefert 2002](https://www.ncbi.nlm.nih.gov/pubmed/12004117)

#### Learning objectives  
Comment on the emergence of microbial life and the evolution of Earth's systems. 

#### General questions  
*Indicate the key events in the evolution of Earth systems at tic marks on the time series.  
*Describe the dominant physical and chemical characteristics of Earth systems at waypoints.

**4.6 Ga**  Solar system formed. The inner planets received water vapour and carbon from meteorite collisions.  
**4.5 Ga**  Moon formation. The Earth's spin and tilt was established.  
**4.4 Ga**  Oldest minerals (zircons) on Earth were found in Western Australia.  
**4.1 Ga**  First evidence of life found in carbon isotopes stored in zircons.  
**4.0 Ga**  Oldest rock (Acasta gneiss) found in Northwestern Canada.  
**3.8 Ga**  Up until now, meteor impacts shaped the seawater chemistry. Emergence of life from Isua belt in southwest Greenland.  
**3.5 Ga**  Rubisco signiture signifies global oygenic photosynthesis and evolution of cyanobacteria.  
**3.2 Ga**  Microfossils in a volcanogenic sulphide deposit in Western Australia suggests life existed on mid-ocean ridges and microbial life was widespread in the mid-Archaean.  
**3.0 Ga**  Abundant evidence of life present in stromatolites.  
**2.7 Ga**  Great oxidation event.  
**2.5 Ga**  Red rock beds found, signify oxidation events.  
**1.7 Ga**  Emergence of eukaryotes.  
**1.1 Ga**  Snowball Earth occured.  
**540 Ma**  Cambrian explosion occured, increased diversity and complexity of organisms.  
**250 Ma**  Permian extinction caused 95% of species to go extinct.  
**65 Ma**  Cretaceous and Paleogene Extinction caused dinosaurs to go extinct.

Describe the dominant physical and chemical characteristics of Earth systems at he following waypoints:  
**Hadean**    
The sun was a third as bright as it was today. CO2 levels were high and the planet was still young and hot, full of molten rock and lava.  
**Archean**    
The atmosphere was methane rich. Emergence of cyanobacteria meant photosynthesize began and the atmosphere started to become oxygenated.  
**Proterozoic**    
Further oxidation events gave rise to complex eukaryotes. Glaciation events happeneed due to decrease in greenhouse effects.  
**Phanerozoic**    
Cambrian explosion created diverse organisms, but extinction events wiped most out.

### Problem Set 2

[Falkowski, P., Fenchel, T., & Delong, E., 2008](https://www.ncbi.nlm.nih.gov/pubmed/12004117)

#### Learning Objectives
Discuss the role of microbial diversity and formation of coupled metabolism in driving global biogeochemical cycles.

#### Specific Questions:

* What are the primary geophysical and biogeochemical processes that create and sustain condiitons for life on Earth? How do abiotic versus biotic processes vary with respect to matter and energy transformation and how are they interconnected?  
Geophysical processes include tectonic and photochemical processes, supplying substrates and removing products, allowing chemical bonds to form and break in cyclical manner. Biogeochemical reactions are based on H, C, N, O, S, and P building blocks, which are driven by microbially catalyzed redox reactions. Abiotic processes are acid-base and biotic processes are driven by redox reactions.

* Why is Earth's redox state considered an emergent property?  
An emergent property is one that the whole system has but the individual members of the system do not have. Feebacks between the evolution of microbial metabolic and geochemical processes create the average redox condition of the oceans and atmosphere, but individual organisms themselves or sole geochemical processes cannot establish the redox condition alone.

* How do reversible electron transfer reactions give rise to element and nutrient cycles at different ecological scales? What strategies do microbes use to overcome thermodynamic barriers to reversible electron flow?  
Reversible electron transfer reactions are possible due to such electron transfer reactions tending towards equilibrium. If the concentration of substrate is too low, the equilibrium will shift and produce more of the limiting substrate. Microbes are symbiotic in that one microbe may provide energy and/or nutrients for another in order for both to overcome thermodynamic barriers to reversible electron flow.

* Using information provided in the text, describe how the nitrogen cycle partitions between different redox "niches" and microbial groups. Is there a relationship between the nitrogen cycle and climate change?  
Nitrogen fixation of N2 to NH4 is required in order for organisms to synthesize nucleic acids and proteins. In the presence of O2, nitrogenase is inhibited and NH4 is oxidized to NO2 then NO3. In the absence of O2, either NO2 or NO3 acts as electron acceptor to produce N2. The nitrogen cycle may cause CO2 levels to drop by converting it to organic matter.

* What is the relationship between microbial diversity and metabolic diversity and how does this relate to the discovery of new protein families from microbial community genomes?  
Microbial diversity allows for catalysis of reversible metabolic pathways. For example, nitrogen fixation is catalyzed by the extremely conserved heterodimeric enzyme complex, nitrogenase, which is inhibited by oxygen. A specific group of bacteria is required to oxidize NH4 to nitrate by oxidizing ammonia to nitrite first, before oxidization to nitrate by a different set of nitrifying bacteria.

* On what basis do the authors consider microbes the guardians of metabolism?  
Dispersal of the core planetary gene set by either vertical or horizontal gene transfer to allow a variety of organisms to guard metabolism. Environmental selection on microbial phenotype leads to evolution of boutique genes that protect the metabolic pathway. The metabolic pathway has a strong chance of surviving in at least one taxon. These taxonomic units may go extinct and evolve but the core machines survive.

### Evidence worksheet 03 

[Rockstrom *et al* 2009](https://www.ncbi.nlm.nih.gov/pubmed/19779433)

#### Learning objectives
Evaluate human impacts on the ecology and biogeochemistry of Earth systems. 

#### General questions
* What were the main questions being asked?  
How far can human development go without causing detrimental or catastrophic effects to the environment? "What are the planetary boundaries"" that define a safe living space on Earth? How many of these boundaries have we already overstepped and what should humans do about it?

* What were the primary methodological approaches used?  
The primary methodological approaches include proposing a framework that contains "planetary boundaries", which are thresholds for each of 9 planetary systems characterized by parameters including climate change and biodiversity loss, that represent the safe operating space for humanity. These thresholds are  critical values taken from literature and are compared and assessed against current values. 

* Summarize the main results or findings.  
The 3 planetary systems of climate change (measured by carbon dioxide concentration), rate of biodiversity loss (measured by extinction rate), and nitrogen cycle (amount of N2 removed from atmosphere for human use) were found to have crossed the proposed planetary boundary. Other system boundaries such as ocean acidification and phosphorus cycle are close to being overstepped.

* Do new questions arise from the results?  
Are there other systems and processes other than the 9 proposed that are important to consider when assessing the impacts of human activity? Could some proposed systems be already overstepped if using different criteria to assess critical values? 

* Were there any specific challenges or advantages in understanding the paper (*e.g.* did the authors provide sufficient background information to understand experimental logic, were methods explained adequately, were any specific assumptions made, were conclusions justified based on the evidence, were the figures or tables useful and easy to understand)?  
The paper provided an insightful view of the state of the anthropocene and provided a good table explicitly showing how each system is assessed and which boundaries are crossed already. However, the authors noted that some boundary thresholds are "merely their best guess", and a more robust method to quantify these thresholds may result in different findings.

### Writing Assessment 01

Microbial life can easily live without us; we, however, cannot survive without the global catalysis and environmental transformations it provides. Microbes have been in existence for much longer than humans have, but with industrialization and other human activities, the human footprint on Earth's environment in our short stint is sizeable enough to alter processes that the sheer number of microbes have established for billions of years. It is the abundance of microbes that exist on Earth along with their high degree of genetic diversity that makes them necessary in sustaining life. Additionally, microbes play a major role in the nitrogen biogeochemical cycle, and with nitrogen being one of the most essential requirements for life to occur, survival without microbes would be challenging. Furthermore, the evolution of complex life such as humans would not have occurred had early microbes not orchestrated pivotal events in Earth's history such as oxygenation events.

Microbes are very abundant on Earth. There are 4-6 x 1030 prokaryotic cells on Earth which together contain 350-550 Pg of carbon, which is 60-100% the cellular carbon found in plants (Whitman et al., 1998). This substantial number of microbes is distributed in major habitats including aquatic habitats, soil, terrestrial subsurface, as well as marine sediments (Whitman et al., 1998). In addition, they have an average cellular production rate of 1.7 x 1030 cells/yr (Whitman et al., 1998). The significance of such a presence of microbes in the world with rapid proliferation is that they have a much higher potential to accumulate mutations and thus to acquire genetic diversity (Whitman et al., 1998). This is important for global catalysis and environmental transformations because without a high degree of genetic diversity, microbes would not be able to adapt to external stimuli and stresses optimally enough to continue powering Earth's biogeochemical cycles efficiently. Some argue that humans are capable of or even obligated to take on an interventionist and managerial role to guide the Earth's ecosystem, but this would require a world of constant intervention in order to fix problems caused by previous interventions (Achenbach, 2012). If humans had developed chemical engineering technologies to command biogeochemical processes at large scale, it would likely require countless iterations to address flaws and optimize processes. It is at the very least challenging to artificially create these biogeochemical processes that microbes have shaped with their highly adaptive potential for billions of years.

Nitrogen is the 5th most abundant element in the solar system and necessary for nucleic acid and protein synthesis, and thus to sustain life. In the atmosphere, however, it is in its inert N2 form and must be fixated to ammonium by microbes through a redox reaction (Canfield et al., 2010). Eukaryotes such as legumes and termites also fixate nitrogen, but only in symbiotic association with nitrogen-fixing prokaryotes, signifying the foundational role of prokaryotes in fixating nitrogen (Canfield et al., 2010). Human activity in the 20th century such as industrial processes, agriculture, and burning fossil fuels have all contributed to the rate of nitrogen fixation, doubling the rate and accounting for 45% of total fixed nitrogen annually produced (Canfield et al., 2010). If microbes did not fixate nitrogen and denitrify, nitrogen flux due to human activity would result in much more fixation than denitrification. In fact, the use of fertilizers in agriculture results in fertilizer wash out due to crop inefficiencies which then go on to be converted and leached into rivers and lakes as NO3-, creating hypoxic zones around the world (Canfield et al., 2010). Clearly, not only can we not live without microbe mediation of the nitrogen biogeochemical cycle, but we would also inflict pollution.

Microbial life has been widespread since the mid-Archaean. Had early microbes not induced important checkpoints during Earth's development, complex organisms such as humans would never have arisen (Nisbet and Sleep, 2003). For example, the evolution of marine cyanobacteria predates 2.7 Ga and is evidence of global oxygenic photosynthesis (Nisbet and Sleep, 2003). In fact, a sharp oxygen spike at 2.2 Ga possibly supported the development of complex eukaryotes.

The abundance of microbes and their high degree of genetic diversity that makes them necessary in sustaining life. They also play a major role in the nitrogen biogeochemical cycle in sustaining life. Also, the evolution of complex life would not have occurred had early microbes not shaped early Earth. With the intelligence of mankind comes consequences to our environment. Having only been in existence for a minute fraction of time compared to the age of the Earth, humans have shown they are capable of impacting the global ecosystem in yet an even shorter amount of time. Over the past few centuries, the actions of humans such as industrialization and agriculture have influenced if not resulted in climate change, biodiversity loss, and interference with the nitrogen cycle, all of which have put the Earth at risk for detrimentally irreversible change (Rockstrom et al., 2009). Microbes may not be greatly affected by these changes due to their adaptability, but humans will certainly be affected. We must recognize that microbes are essential in maintaining a viable ecosystem and be wary of the environmental impacts of our actions. 

### Module 01 references

[Achenbach, J. (2012). Spaceship Earth: A new view of environmentalism. The Washington Post.](https://www.washingtonpost.com/national/health-science/spaceship-earth-a-new-view-of-environmentalism/2011/12/29/gIQAZhH6WP_story.html?noredirect=on&utm_term=.8df16597e0fa)  
[Canfield, D., Glazer, A., & Falkowski, P. (2010). The Evolution and Future of Earth's Nitrogen Cycle. Science, 330(6001), 192-196.](https://www.ncbi.nlm.nih.gov/pubmed/20929768)  
[Falkowski, P., Fenchel, T., & Delong, E. (2008). The Microbial Engines That Drive Earth's Biogeochemical Cycles. Science, 320(5879), 1034-1039.](https://www.ncbi.nlm.nih.gov/pubmed/18497287)  
[Kasting, J. (2003). Life and the Evolution of Earth's Atmosphere. Science, 296(5570), 1066-1068.](https://www.ncbi.nlm.nih.gov/pubmed/12004117)  
[Nisbet, E., & Sleep, N. (2001). The habitat and nature of early life. Nature, 409(6823), 1083-1091.](https://www.nature.com/articles/35059210)  
[Rockström, J., Steffen, W., Noone, K., Persson, Å., Chapin, F., & Lambin, E. et al. (2009). A safe operating space for humanity. Nature, 461(7263), 472-475.](https://www.nature.com/articles/461472a)  
[Whitman WB, Coleman DC, and Wiebe WJ. 1998. Prokaryotes: The unseen majority. Proc Natl Acad Sci USA. 95(12):6578-6583.](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC33863/)



#Module 2

## Module 2 portfolio check
The first of your second level headers (##) is to be used for the portfolio content checks. The Module 01 portfolio check has been built for you directly into this template, but will also be available as a stand-alone markdown document available on the MICB425 GitHub so that you know what is required in each module section in your portfolio. The completion status and comments will be filled in by the instructors during portfolio checks when your current portfolios are pulled from GitHub.

* Evidence worksheet_04
    + Completion status:
    + Comments: Incomplete
* Problem Set_03
    + Completion status:
    + Comments: Incomplete
* Writing assessment_02
    + CANCELED
* Additional Readings
    + Completion status: Incomplete
    + Comments: Missing links. Out of order with Module_03 

### Problem set_03 - Metagenomics: Genomic Analysis of Microbial Communities
[Wooley *et al* 2010](https://www.ncbi.nlm.nih.gov/pubmed/?term=A+Primer+on+Metagenomics+wooley)  
[Solden, Lloyd, and Wrighton 2016](https://www.ncbi.nlm.nih.gov/pubmed/27196505)  
[JGI GOLD Database](https://gold.jgi.doe.gov/projects)  
[Solden, Lloyd & Wrighton, 2016](https://www.sciencedirect.com/science/article/pii/S1369527416300558?via%3Dihub)  
[JGI GOLD Database](https://gold.jgi.doe.gov/projects)  
[Krause *et al* 2008](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2367736/)  
[Beaulaurier *et al* 2018](https://www.nature.com/articles/nbt.4037)  
[Taupp, Mewis & Hallam, 2011](https://www-sciencedirect-com.ezproxy.library.ubc.ca/science/article/pii/S0958166911000371)

#### Learning objectives: 
Specific emphasis should be placed on the process used to find the answer. Be as comprehensive as possible e.g. provide URLs for web sources, literature citations, etc.  

#### Specific Questions:
* How many prokaryotic divisions have been described and how many have no cultured representatives (microbial dark matter)?  
At least 89 bacterial and 20 archael phyla have been recognized by small subunit ribosomal RNA databases, although there are likely more. However, only 0.1-1% of microorganisms have been cultured, leaving many prokaryotic phyla with no cultured representatives.

* How many metagenome sequencing projects are currently available in the public domain and what types of environments are they sourced from?  
According to the Joint Genome Institute, there are 197,062 metagenome sequencing projects are currently available. The projects are sourced from different environments including soils, marine sediments, seawater, and feces.

* What types of on-line resources are available for warehousing and/or analyzing environmental sequence information (provide names, URLS and applications)?  
[MEGAN](ab.inf.uni-tuebingen.de/software/megan/) is a binning software that can also be used to compare the OTU composition of two or more frequency-normalized samples. [MG-RAST](https://metagenomics.anl.gov/) provides comparisons between functional and sequence-based analysis for uploaded samples, whereas [IMG/M](https://img.jgi.doe.gov/m/) provides similar analysis for metagenomes that exist on the site. [CAMERA](http://camera.calit2.net) performs BLAST on sequences against metagenomic datasets and is an archive for metagenomic datasets generated by marine microbial research.

* What is the difference between phylogenetic and functional gene anchors and how can they be used in metagenome analysis?   
Phylogenetic gene anchors use conserved markers to predict the taxonomic origins of environmental genomic fragments, while functional gene anchors use such information to acquire information about metabolic pathways. In metagenome analyses, phylogenetic gene anchors can predict the evolutionary relationship between organisms while functional gene anchors uncover metabolic pathways of species, both of which help construct a phylogenetic tree.

* What is metagenomic sequence binning? What types of algorithmic approaches are used to produce sequence bins? What are some risks and opportunities associated with using sequence bins for metabolic reconstruction of uncultivated microorganisms?  
Metagenomic sequence binning is when sequences are grouped and assigned to taxonomical units. Algorithmic approaches to group sequences include sequence composition, species abundance, chromosome organization, or DNA methylation signatures. Sequence bins can help with discovering genes that constitute metabolic pathways, but it cannot fully distinguish closely related species and strains due to incomplete covereage of genome sequences.

* Is there an alternative to metagenomic shotgun sequencing that can be used to access the metabolic potential of uncultivated microorganisms? What are some risks and opportunities associated with this alternative?  
An alternative to metagenomic shotgun sequencing is functional metagenomic screens, which looks at phenotypic properties of organisms to construct metabolic pathways and gene discovery. Functional screens allow for gene models to be validated and protein expression to be linked to specific parts of metabolic pathways. However, knowledge and availability of environmental parameter data such as sampling conditions is necessary to select appropriate screening targets and substrates.

### Evidence Worksheet 04
[Martinez *et al* 2007](https://www.ncbi.nlm.nih.gov/pubmed/17372221)

#### Learning objectives
* Discuss the relationship between microbial community structure and metabolic diversity  
*	Evaluate common methods for studying the diversity of microbial communities.  
*	Recognize basic design elements in metagenomic workflows. 

#### General questions
* What were the main questions being asked?  
What are the specific functions and physiological roles of marine microbial proteorhodopsins? What is the extent of *in vivo* PR Photosystem expression in a marine picoplankton genomic library? What do genotypic and phenotypic analysis of the PR Photosystem show? What is the effect of light activated proton translocation on metabolic pathways?

* What were the primary methodological approaches used?  
A fosmid library from planktonic DNA sample 10m below surface was used for the screen. To screen for PR expression, high-density colony microarrays were used with LB agar supplemented with L-arabinose. Colonies were visually inspected for orange/red colour. Fosmid clones to be sequenced were subjected to Tn5 insertion, and sequences were assembled using Sequencher version 4.5, annotated with FGENESB and Artemis version 6. Carotenoid extracts were concentrated by evaporation with N2 and performed under darkness to minimize photooxidation.

* Summarize the main results or findings.  
Fosmid library screening revealed an increase in copy number that significantly enhanced sensitivity of phenotypic detection. Two genetically distinct recombinants expressed an orange colour under L-arabinose treatment and expressed a small cluster of genes encoding the complete PR-based photosystem. Heterologous expression of 6 genes involved with PR photosystem generated a fully functional photosystem that enabled photophosphorylation in recombinant *E. coli* cells after light exposure.

* Do new questions arise from the results?  
What is the effect of lateral gene transfer of PR photosystem genes in microbes and how does this affect the function of the photosystem in different species? To what extent does photooxidation hinder the photophosphorylation and thus the functionality of the PR photosystem?

* Were there any specific challenges or advantages in understanding the paper (*e.g.* did the authors provide sufficient background information to understand experimental logic, were methods explained adequately, were any specific assumptions made, were conclusions justified based on the evidence, were the figures or tables useful and easy to understand)?  
The authors provided sufficient background information regarding proteorhodopsins, their abundance in marine habitats, and PR photosystems for understanding of the experiments performed. Methods were explained clearly and concisely, figures were useful for understanding, and conclusions were justified based on experimental results.

### Module 2 References

[Beaulaurier, J., Zhu, S., Deikus, G., Mogno, I., Zhang, X., & Davis-Richardson, A. et al. (2017). Metagenomic binning and association of plasmids with bacterial host genomes using DNA methylation. Nature Biotechnology, 36(1), 61-69.](https://www.nature.com/articles/nbt.4037)  
[JGI GOLD | Projects. (2018). Gold.jgi.doe.gov.](Gold.jgi.doe.gov)  
[Krause, L., Diaz, N., Goesmann, A., Kelley, S., Nattkemper, T., & Rohwer, F. et al. (2008). Phylogenetic classification of short environmental DNA fragments. Nucleic Acids Research, 36(7), 2230-2239.](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2367736/)  
[Martinez, A., Bradley, A., Waldbauer, J., Summons, R., & DeLong, E. (2007). Proteorhodopsin photosystem gene expression enables photophosphorylation in a heterologous host. Proceedings Of The National Academy Of Sciences, 104(13), 5590-5595.](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC1838496/)  
[Solden, L., Lloyd, K., & Wrighton, K. (2016). The bright side of microbial dark matter: lessons learned from the uncultivated majority. Current Opinion In Microbiology, 31, 217-226.](https://www.sciencedirect.com/science/article/pii/S1369527416300558?via%3Dihub)  
[Taupp, M., Mewis, K., & Hallam, S. (2011). The art and design of functional metagenomic screens. Current Opinion In Biotechnology, 22(3), 465-472.](https://www-sciencedirect-com.ezproxy.library.ubc.ca/science/article/pii/S0958166911000371)  
[Wooley, J., Godzik, A., & Friedberg, I. (2010). A Primer on Metagenomics. Plos Computational Biology, 6(2), e1000667.](https://www.ncbi.nlm.nih.gov/pubmed/?term=A+Primer+on+Metagenomics+wooley)



# Module 3

## Module 3 Portfolio Check

* Evidence worksheet_05
    + Completion status: X
    + Comments:
* Problem set_04
    + Completion status:
    + Comments:
* Writing Assessment_03
    + Completion status:
    + Comments:
* Additional Readings
    + Completion status:
    + Comments 
* CATME account setup and survey
    + Completion status: X
    + Comments: 
* CATME interim group assessment
    + Completion status: X
    + Comments:
* Project 1
    + Report (80%): 
    + Participation (20%):

### Evidence Worksheet 05
[Welch *et al* 2002](https://www.ncbi.nlm.nih.gov/pubmed/12471157)

#### Learning objectives
* Evaluate the concept of microbial species based on environmental surveys and
cultivation studies.  
* Explain the relationship between microdiversity, genomic diversity and metabolic potential.  
* Comment on the forces mediating divergence and cohesion in natural microbial
communities. 

#### General questions
* What were the main questions being asked?  
How do the 2 pathogenic and 1 non-pathogenic E. coli strains differ in their pathogenecity of intestinal diseases? From what lineages did pathogenic E. coli arise from? How does lateral gene transfer lead to new uropathogenic E. coli strains? What is the core gene set as opposed to the accessory gene set in each strain?

* What were the primary methodological approaches used?  
Whole genome libraries prepared from genomic DNA and sequenced by Sanger sequencing. Genome sequence was annotated using MAGPIE and ORFs defined using GLIMMER. Predicted proteins were searched with BLAST. Shotgun library sequencing and PCR of CFT073 genome were performed. 

* Summarize the main results or findings.  
Assembly and alignment of CFT073 DNA fragment library to MG1655 and EDL933 showed >70% ORFs previously identified as unique to MG1655 and EDL933 are replaced with CFT073 specific genes. Acquisition of islands resulted in ability to infect urinary tracts and bloodstream, while colonizing the intestine and evading host defenses. Acquired genes promote colonization of specific regions of the intestine and new modes of interaction with host tissue, producing clincally distict variations of gastrointestinal disease.

* Do new questions arise from the results?  
Do the strain specific genes allow its mechanism of infection to differ from other strains? What are the factors (eg. environmental) that influence the rate of gene acquisition? What are some other implications of island acquisition other than ability to infect bloodstream and evade host defenses?

* Were there any specific challenges or advantages in understanding the paper (*e.g.* did the authors provide sufficient background information to understand experimental logic, were methods explained adequately, were any specific assumptions made, were conclusions justified based on the evidence, were the figures or tables useful and easy to understand)?  
Understanding this paper require several read throughs as it was heavily genetics based research. Background info and methods were explained adequately, however more informative figures would offer a better visual interpretation of the results.

### Problem set_04

#### Learning objectives: 

Gain experience estimating diversity within a hypothetical microbial community

#### Outline:

In class Day 1:

1. Define and describe species within your group's "microbial" community.
2. Count and record individuals within your defined species groups.
3. Remix all species together to reform the original community.
4. Each person in your group takes a random sample of the community (*i.e.* devide up the candy).

Assignment:

5. Individually, complete a collection curve for your sample.
6. Calculate alpha-diversity based on your original total community and your individual sample.

In class Day 2:

7. Compare diversity between groups.

#### Part 1: Description and enumeration
Obtain a collection of "microbial" cells from "seawater". The cells were concentrated from different depth intervals by a marine microbiologist travelling along the Line-P transect in the northeast subarctic Pacific Ocean off the coast of Vancouver Island British Columbia. 

Sort out and identify different microbial "species" based on shared properties or traits. Record your data in this Rmarkdown using the example data as a guide.

Once you have defined your binning criteria, separate the cells using the sampling bags provided. These operational taxonomic units (OTUs) will be considered separate "species". This problem set is based on content available at [What is Biodiversity](http://cnx.org/content/m12179/latest/).

```{r message=FALSE}
#To make tables
library(kableExtra)
library(knitr)
#To manipulate and plot data
library(tidyverse)
library(vegan)
```

Load in data.

```{r}
# Data shared by Lucas Chang
# Species are defined as different candy types based on shape and structure.

#Original Sample
candytable = data.frame(
  number = c(1,2,3,4,5,6,7,8,9,10,11,12,13),
  name = c("Twizzler", "OG Gummy", "Wine Gummy", "Sugar Gummy", "Sugar Swirls", "Sugar Bottles", "Octopi", "Mike-Ike", "Sphere Candy", "SKittles", "M&Ms", "Kisses", "Lego"),
  characteristics = c("red string", "gummy bear", "wine gummy bear", "sugar gummy bear", "sugar coated swirls", "sugar coated bottle", "sugar coated octopi", "ovoid chewy", "spherical chewy", "flat hardshell chocolate", "bulky hardshell chocolate", "flat-bottom teardrop chocolate", "lego-like hard chewy"), 
  occurences = c(7,16,1,2,1,1,4,25,6,26,41,1,1)
)

candytable %>% 
  kable("html") %>%
  kable_styling(bootstrap_options = "striped", font_size = 10, full_width = F)

#Community
OGcommunity_data1 = data.frame(
  number = c(1,2,3,4,5,6,7,8,9,10,11,12,13),
  name = c("Strings", "OG Gummy", "Wine Gummy", "Sugar Gummy", "Sugar Swirls", "Sugar Bottles", "Octopi", "Mike-Ike", "Sphere Candy", "SKittles", "M&Ms", "Kisses", "Lego"),
  characteristics = c("red string", "gummy bear", "wine gummy bear", "sugar gummy bear", "sugar coated swirls", "sugar coated bottle", "sugar coated octopi", "ovoid chewy", "spherical chewy", "flat hardshell chocolate", "bulky hardshell chocolate", "flat-bottom teardrop chocolate", "lego-like hard chewy"), 
  occurences = c(14,101,8,3,3,3,6,131,19,197,242,16,18)
)

OGcommunity_data1 %>% 
  kable("html") %>%
  kable_styling(bootstrap_options = "striped", font_size = 10, full_width = F)

# The collection of "organisms" samples all species. If the sample size was very small, then many would be missed.
```

#### Part 2: Collector's curve
To help answer the questions raised in Part 1, you will conduct a simple but informative analysis that is a standard practice in biodiversity surveys. This analysis involves constructing a collector's curve that plots the cumulative number of species observed along the y-axis and the cumulative number of individuals classified along the x-axis. This curve is an increasing function with a slope that will decrease as more individuals are classified and as fewer species remain to be identified. If sampling stops while the curve is still rapidly increasing then this indicates that sampling is incomplete and many species remain undetected. Alternatively, if the slope of the curve reaches zero (flattens out), sampling is likely more than adequate.

To construct the curve for your samples, choose a cell within the collection at random. This will be your first data point, such that X = 1 and Y = 1. Next, move consistently in any direction to a new cell and record whether it is different from the first. In this step X = 2, but Y may remain 1 or change to 2 if the individual represents a new species. Repeat this process until you have proceeded through all cells in your collection.

```{r}
candytable2 = data.frame(
  Number = c(1,2,3,4,5,6,7,8,9,10,11,12,13),
  Occurences = c(7,16,1,2,1,1,4,25,6,26,41,1,1)
)

rarecurve(candytable2, step = 1, xlab = "Cumulative Number of Classified Species", ylab = "Cumulative Number of Observed Species", label = TRUE)
```

* Does the curve flatten out? If so, after how many individual cells have been collected?  
The curve flattens out after about 18 individual cells have been collected.

* What can you conclude from the shape of your collector's curve as to your depth of sampling?  
The degree of sampling was sufficient to achieve adequate representation of the population.

#### Part 3: Diversity estimates (alpha diversity)  
Using the table from Part 1, calculate species diversity using the following indices or metrics.

##### Diversity: Simpson Reciprocal Index

$\frac{1}{D}$ where $D = \sum p_i^2$

$p_i$ = the fractional abundance of the $i^{th}$ species

```{r}
# Sample Species Diversity
community = candytable %>% 
  # Select only name and count columns
  select(name, occurences) %>%
  # Spread into taxa as columns format
  spread(name, occurences) %>% 
  # Convert to phyloseq OTU table data type
  otu_table(taxa_are_rows = FALSE) 

# Create a random subsample (rarefy) the data to 100 observations
subsample = as.data.frame(rarefy_even_depth(community, sample.size=100, replace=FALSE, rngseed=762))
# 1OTUs were removed because they are no longer present in any sample after random subsampling

subsample %>% 
  kable("html") %>%
  kable_styling(bootstrap_options = "striped", font_size = 10)

cspecies1 = 1/100
cspecies2 = 7/100
cspecies3 = 13/100
cspecies4 = 1/100
cspecies5 = 4/100
cspecies6 = 3/100
cspecies7 = 1/100
cspecies8 = 26/100
cspecies9 = 24/100
cspecies10 = 1/100
cspecies11 = 18/100
cspecies12 = 1/100

# Simpson Reciprocal Index  
1 / (cspecies1^2 + cspecies2^2 + cspecies3^2 + cspecies4^2 + cspecies5^2 + cspecies6^2 + cspecies7^2 + cspecies8^2 + cspecies9^2 + cspecies10^2 + cspecies11^2 + cspecies12^2)

# Community Species Diversity
community1 = OGcommunity_data1 %>% 
  select(name, occurences) %>%
  spread(name, occurences) %>% 
  otu_table(taxa_are_rows = FALSE) 

cspecies1 = 14/761
cspecies2 = 101/761
cspecies3 = 8/761
cspecies4 = 3/761
cspecies5 = 3/761
cspecies6 = 3/761
cspecies7 = 6/761
cspecies8 = 131/761
cspecies9 = 19/761
cspecies10 = 197/761
cspecies11 = 214/761
cspecies12 = 16/761
cspecies13 = 18/761

# Simpson Reciprocal Index  
1 / (cspecies1^2 + cspecies2^2 + cspecies3^2 + cspecies4^2 + cspecies5^2 + cspecies6^2 + cspecies7^2 + cspecies8^2 + cspecies9^2 + cspecies10^2 + cspecies11^2 + cspecies12^2 + cspecies13^2)
```
The higher the value is, the greater the diversity. The maximum value is the number of species in the sample, which occurs when all species contain an equal number of individuals. Because the index reflects the number of species present (richness) and the relative proportions of each species with a community (evenness), this metric is a diveristy metric. Consider that a community can have the same number of species (equal richness) but manifest a skewed distribution in the proportion of each species (unequal evenness), which would result in different diveristy values.

* What is the Simpson Reciprocal Index for your sample?  
5.482456

* What is the Simpson Reciprocal Index for your original total community?  
5.114509


##### Richness: Chao1 richness estimator
Another way to calculate diversity is to estimate the number of species that are present in a sample based on the empirical data to give an upper boundary of the richness of a sample. Here, we use the Chao1 richness estimator. 

$S_{chao1} = S_{obs} + \frac{a^2}{2b})$

$S_{obs}$ = total number of species observed
a = species observed once
b = species observed twice or more

Sample:  
$S_{obs}$ = total number of species observed
a = 5
b = 7
```{r}
12 + 5^2/(2*7)
```

Original Total Community  
$S_{obs}$ = total number of species observed
a = 0
b = 13
```{r}
13 + 0^2/(2*13)
```

* What is the chao1 estimate for your sample?
13.78571

* What is the chao1 estimate for your original total community?
13

#### Part 4: Alpha-diversity functions in R

First, we must remove the unnecesary data columns and transpose the data so that `vegan` reads it as a species table with species as columns and rows as samples (of which you only have 1).

```{r}
OGcommunity_data1_diversity = 
  candytable %>% 
  select(name, occurences) %>% 
  spread(name, occurences)

OGcommunity_data1_diversity
```

Then we can calculate the Simpson Reciprocal Index using the `diversity` function.
```{r}
#Sample Data
diversity(subsample, index="invsimpson")

#Community Data
diversity(OGcommunity_data1_diversity, index="invsimpson")
```

And we can calculate the Chao1 richness estimator (and others by default) with the the `specpool` function for extrapolated species richness. This function rounds to the nearest whole number so the value will be slightly different that what you've calculated above.
```{r}
#Sample Data
specpool(subsample)

#Community Data
specpool(OGcommunity_data1_diversity)
```

For your sample:
* What are the Simpson Reciprocal Indices for your sample and community using the R function?  
Sample: 5.482456
Community: 5.204301

* What are the chao1 estimates for your sample and community using the R function?
    + Verify that these values match your previous calculations.

Sample:
    Species chao chao.se jack1 jack1.se jack2 boot boot.se n
All      12   12       0    12        0    12   12       0 1

Community:
    Species chao chao.se jack1 jack1.se jack2 boot boot.se n
All      13   13       0    13        0    13   13       0 1

Simpson Reciprocal Values for sample match previous calculations but not for community (R function generates higher Simpson Value than by hand), and chao1 estimates are different when calculating by hand vs. by R function specpool

#### Part 5: Concluding activity

* How does the measure of diversity depend on the definition of species in your samples?  
Measure of diversity depends on how loose the definition of species is. A stricter definition of species (eg. sorting candies by shape AND colour AND size) would yield a greater number of species and higher Simpson Value.

* Can you think of alternative ways to cluster or bin your data that might change the observed number of species?  
Some other ways the dataset could be classified is by colour or size. This could increase the observed number of species, leading to a higher measure of diversity and higher Simpson Value.

* How might different sequencing technologies influence observed diversity in a sample?  
Sequencing technologies that have stricter guidelines and definitions for a species would yield more observed species and thus a higher observed diversity in a sample. The use of different primers for 16S rRNA gene amplification will also affect the observed number of species and thus observed diversity. 

### Writing Assessment 03

Defining microbial species has been a challenge in the scope of microbial ecology due to differences in phylogenetic groups and population diversity, as well as hurdles associated with understanding gene content and ecological distinctiveness (Konstantinidis, Ramette & Tiedje, 2006). In the past, species definitions have been proposed such as a collection of strains that are characterized by at least one diagnostic phenotypic trait and whose purified DNA molecules show at least 70% cross-hybridization (Konstantinidis, Ramette & Tiedje, 2006). Such definitions have its pitfalls, such as challenges in experimental execution and too much phenotypic variation within a defined species. (Konstantinidis, Ramette & Tiedje, 2006). Recent advances in genomics and DNA sequencing technology have allowed for single nucleotide differences to be detectable in marker-gene sequencing, establishing stringent new methods to characterize different taxonomic groups (Callahan, McMurdie, and Holmes, 2017). Additionally, lateral gene transfer has been evident in metagenomic analyses of bacterial artificial chromosomes and shotgun small-insert libraries, demonstrating vast metabolic diversity and exchange of genomic information (Sogin et al., 2006). Lateral gene transfer allows metabolic pathways and functions to be acquired from other prokaryotes and genome architecture to be restructured, which further adds uncertainty to the definition of a microbial species. Lateral gene transfer also influences the maintenance of global biogeochemical cycles through transfer of metabolic genes such as proteorhodopsin genes, allowing metabolic pathways to be acquired by different groups (Martinez et al., 2007). The importance in having a clear definition of a microbial species depends on the questions being addressed and the outcomes expected. For some purposes, a clear definition of a microbial species may be beneficial to find a solution for the problem at hand, such as the diagnoses and treatment of infections. However, from a basic research perspective, given the current challenges faced in defining a species with certainty, a flexible definition of a species may be better for advancing our understanding of genomics and ultimately lead to better iterations of the species definition.

Understanding of genetic differences in microbes and how they relate to function, especially within closely related microbes, is important to define a microbial species with clarity. In other words, a high resolution of genome sequencing is required. One of the most influential set of criteria for defining a microbial species, described roughly three decades earlier, is a collection of strains that are characterized by at least one diagnostic phenotypic trait and whose DNA molecules show at least 70% cross-hybridization (Wayne et al., 1987). Issues arose, however, from such a definition, and only ~4500 species were described in contrast to over one million eukaryotic species defined within the same time frame (Konstantinidis, Ramette & Tiedje, 2006). Challenges such as implementing DNA cross-hybridization experiments, applicability in environmental surveys, and too much phenotypic variation within a defined species were attributed to the small number of species described in comparison to eukaryotes, even though microbes have been adapting and evolving for a much longer time (Konstantinidis, Ramette & Tiedje, 2006). It was clear that the microbial species definition had to be refined by better understanding of genetic differences between microbes. With the improvement of DNA sequencing technology and increasing number of genome datasets, techniques such as high-throughput sequencing of gene markers have allowed taxonomic profiling of microbial communities (Callahan, McMurdie, and Holmes, 2017). Clusters of reads are created from gene marker studies whose sequences differ by less than a set threshold, also known as operational taxonomic units (OTUs) (Callahan, McMurdie, and Holmes, 2017). However, such dissimilarity thresholds are arbitrary, and stringency has been furthered in methods of clustering such as generating amplicon sequencing variants (ASVs) which distinguish sequence variants differing by as little as one nucleotide have been proposed (Callahan, McMurdie, and Holmes, 2017).

Advances in sequencing technologies have helped establish a stricter species definition, but lateral gene transfer between different groups of microbes remains a challenge. A whole genome analysis of the uropathogenic E.coli CFT073 and similar strains revealed that only 39.2% of proteins were common to all three strains (Welch et al., 2002). Experiments involving codon usage, which is distinctive of lateral gene transfer, revealed 52 of 61 codons in islands open reading frames (ORFs) were significantly used (Welch et al., 2002). Ancestral backbone genes that encoded essential functions such as iron-uptake systems were conserved and shared between strains, but many other genes were independently acquired through horizontal gene transfer processes (Welch et al., 2002). In addition, photoproteins that catalyze proton translocation across the cell membrane called proteorhodopsins have been shown to distribute to the archaeal Thermoplasmatales which reside in the ocean's photic zone along with proteobacteria (Frigaard et al., 2006). Proteorhodopsins are driven by lateral gene transfer mechanisms with its broad phylogenetic diversity; spatial distribution of such widely distributed genes appears to be influenced by environmental stresses rather than the organisms that acquire them (Frigaard et al., 2006). Transfer of genes such as proteorhodopsins also affect global biogeochemical processes and cycles by increasing the prevalence of photosystems and other metabolic pathways in marine environments.

It has been argued whether or not having a definitive definition of a species is necessary. For purposes such as diagnosis of infectious disease agents and establishing regulations for transport and possession of pathogens, having a stringent definition of a microbial species would be beneficial. However, in the aspect of basic research and scientific curiosity, having an open, more fluid definition that changes with new discoveries may be advantageous in advancing the field of genomics, which in return betters the definition itself.

The goal of defining a microbial species has indeed been a challenge in genomics due to lapses in data and understanding of the species concept. Horizontal gene transfer has not made the process easier - metabolic functions and proteins are distributed to other groups of organisms that reside within the same environment, contributing to a large portion of their genome and affecting global biogeochemical cycles. However, with advances in genomics, sequencing data is churned out on a regular basis - marker gene studies generate millions of sequencing reads - which could only raise further questions about the species concept and further fuel our desire to understand. Given the large amounts of genome data produced, the next step is making sense of it all. Bioinformatics has been the forefront of analyzing big data, and as such our knowledge of the species concept will undoubtedly increase in the near future.


### Project 1

Refer to 18315144_Project1 markdown in portfolio repo.

### Module 3 References

[Callahan, B., McMurdie, P., & Holmes, S. (2017). Exact sequence variants should replace operational taxonomic units in marker-gene data analysis. The ISME Journal, 11(12), 2639-2643.](https://www.nature.com/articles/ismej2017119)  
[Frigaard, N., Martinez, A., Mincer, T., & DeLong, E. (2006). Proteorhodopsin lateral gene transfer between marine planktonic Bacteria and Archaea. Nature, 439(7078), 847-850.](https://www.nature.com/articles/nature04435)  
[Konstantinidis, K., Ramette, A., & Tiedje, J. (2006). The bacterial species definition in the genomic era. Philosophical Transactions Of The Royal Society B: Biological Sciences, 361(1475), 1929-1940.](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC1764935/)  
[Martinez, A., Bradley, A., Waldbauer, J., Summons, R., & DeLong, E. (2007). Proteorhodopsin photosystem gene expression enables photophosphorylation in a heterologous host. Proceedings Of The National Academy Of Sciences, 104(13), 5590-5595.](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC1838496/)  
[Sogin, M., Morrison, H., Huber, J., Welch, D., Huse, S., & Neal, P. et al. (2006). Microbial diversity in the deep sea and the underexplored "rare biosphere". Proceedings Of The National Academy Of Sciences, 103(32), 12115-12120.](http://www.pnas.org/content/103/32/12115)  
[Wayne, L. (1987). International committee on systematic bacteriology: Announcement of the report of the ad hoc committee on reconciliation of approaches to bacterial systematics. Zentralblatt Für Bakteriologie, Mikrobiologie Und Hygiene. Series A: Medical Microbiology, Infectious Diseases, Virology, Parasitology, 268(4), 433-434.](http://www.microbiologyresearch.org/docserver/fulltext/ijsem/37/4/ijs-37-4-463.pdf?expires=1524579041&id=id&accname=guest&checksum=393582B3EE17FD71CBC0FCDB56677692)  
[Welch, R., Burland, V., Plunkett, G., Redford, P., Roesch, P., & Rasko, D. et al. (2002). Extensive mosaic structure revealed by the complete genome sequence of uropathogenic Escherichia coli. Proceedings Of The National Academy Of Sciences, 99(26), 17020-17024.](https://www.ncbi.nlm.nih.gov/pubmed/12471157)

# Module 4

## Module 4 Portfolio Check

* CATME final group assessment
    + Completion status: 
    + Comments:
* Project 2
    + Report (80%): 
    + Participation (20%):
    
### Project 2

Refer to 18315144_Project2 markdown in portfolio repo.