---
title: "MICB425 Portfolio - Microbial Ecological Genomics: From Genomes to Biomes, Spring 2018"
author: "Paaksum Wong, 18315144"
date: "version `r format(Sys.time(), '%B %d, %Y')`"
output: 
  html_document:
    toc: true
---
# Module 01
Reserve the first level headings (#) for the start of a new Module. This will help to organize your portfolio in an intuitive fashion.  
**Note: Please edit this template to your heart's content. This is meant to be the armature upon which you build your individual portfolio. You do not need to keep this instructive text in your final portfolio, although you do need to keep module and assignment names so we can identify what is what.**


## Module 01 portfolio check
The first of your second level headers (##) is to be used for the portfolio content checks. The Module 01 portfolio check has been built for you directly into this template, but will also be available as a stand-alone markdown document available on the MICB425 GitHub so that you know what is required in each module section in your portfolio. The completion status and comments will be filled in by the instructors during portfolio checks when your current portfolios are pulled from GitHub.

* Installation check
    + Completion status:
    + Comments:
* Portfolio repo setup
    + Completion status: 
    + Comments:
* RMarkdown Pretty PDF Challenge
    + Completion status:
    + Comments:
* Evidence worksheet_01
    + Completion status:
    + Comments:
* Evidence worksheet_02
    + Completion status:
    + Comments:
* Evidence worksheet_03
    + Completion status:
    + Comments:
* Problem Set_01
    + Completion status:
    + Comments:
* Problem Set_02
    + Completion status:
    + Comments:
* Writing assessment_01
    + Completion status:
    + Comments:
* Additional Readings
    + Completion status:
    + Comments  
    
## Data science Friday  
The remaining second level headers (##) are for separating data science Friday, regular course, and project content. In this module, you will only need to include data science Friday and regular course content; projects will come later in the course.

### Installation check
Third level headers (###) should be used for links to assignments, evidence worksheets, problem sets, and readings, as seen here.  

Use this space to include your installation screenshots.

![Git Bash Terminal](Git Bash Terminal.jpg)
![RStudio Screenshot](RStudio Screenshot.jpg)
![GitHub Homepage](GitHub Homepage.jpg)

### Portfolio repo setup
Detail the code you used to create, initialize, and push your portfolio repo to GitHub. This will be helpful as you will need to repeat many of these steps to update your porfolio throughout the course.

Pushing files to GitHub:
Navigate to portfolio directory by using "cd ~/Documents" and "cd MICB425_portfolio".
"git.add"
"git commit -m "nth commit""
"git push"
Verify on GitHub it was successfully pushed.

### R Markdown Pretty html Challenge
Refer to Assignment 3 markdown file in portfolio.

### ggplot Exercises
```{r}
library(tidyverse)
source("https://bioconductor.org/biocLite.R")
biocLite("phyloseq")
library("phyloseq")
load("phyloseq_object.RData")
metadata=read.table(file="Saanich.metadata.txt", header=TRUE, row.names=1, sep="\t")
ggplot(metadata, aes(x=O2_uM, y=Depth_m))+
  geom_point(shape="square")
ggplot(metadata, aes(x=O2_uM, y=Depth_m))+
  geom_point(size=1)
ggplot(metadata, aes(x=O2_uM, y=Depth_m, size=OxygenSBE_V))+
  geom_point()

#Exercise 1
ggplot(metadata, aes(x=PO4_uM, y=Depth_m))+
  geom_point(shape=17, color="purple")

#Exercise 2
metadata %>% 
  mutate(Temperature_F=Temperature_C*1.8+32) %>% 
  ggplot(aes(x=Temperature_F, y=Depth_m))+
  geom_point()

#gglot with phyloseq
physeq
plot_bar(physeq, fill="Phylum")
physeq_percent=transform_sample_counts(physeq, function(x) 100 * x/sum(x))
plot_bar(physeq, fill = "Phylum")
plot_bar(physeq_percent, fill = "Phylum")+
  geom_bar(aes(fill=Phylum), stat="identity")

#Exercise 3
plot_bar(physeq_percent, fill = "Species")+
  geom_bar(aes(fill=Phylum), stat="identity")+
  labs(title="Species from 10 to 200 m in Saanich Inlet", x="Sample depth", y="Percent relative abundance")

#Faceting
plot_bar(physeq_percent, fill = "Phylum")+
  geom_bar(aes(fill=Phylum), stat="identity")+
  facet_wrap(~Phylum)
plot_bar(physeq_percent, fill = "Phylum")+
  geom_bar(aes(fill=Phylum), stat="identity")+
  facet_wrap(~Phylum, scales = "free_y")+
  theme(legend.position = "none")

#Exercise 4
metadata %>% 
  gather(nutrients, uM, O2_uM, PO4_uM, SiO2_uM, NO3_uM, NH4_uM, NO2_uM) %>% 
  ggplot(aes(Depth_m, uM))+
  geom_point()+
  geom_line()+
  facet_wrap(~nutrients, scales = "free_y")+
  theme(legend.position = "none")

```

## Origins and Earth Systems

### Evidence worksheet 01 

[Whitman *et al* 1998](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC33863/)

#### Learning objectives
Describe the numerical abundance of microbial life in relation to ecology and biogeochemistry of Earth systems. 

#### General questions
* What were the main questions being asked?  
What are the primary habitats of prokaryotes? What is their abundance and total carbon content in each of the main habitats? What is the total nutrient content in these habitats?

* What were the primary methodological approaches used?  
The primary methodological approaches included estimating cellular density, volume, and carbon to estimate the number of prokaryotes in different habitats. Extrapolation of cell densities were used to determine number of subsurface prokaryotes. To determine amount of carbon in prokaryotes, cellular carbon content was assumed to be half the dry weight.

* Summarize the main results or findings.  
The main habitats of prokaryotes and their number of prokaryotes are aquatic habitats (1.2x10^29^), oceanic subsurface (3.5x10^30^), soil (2.6x10^29^), and terrestrial subsurface (0.25-2.5x10^30^). The total number of prokaryotic cells on Earth is 4-6x10^30^ and their total carbon content is 350-550Pg. They also contain 85-130Pg N and 9-14Pg P. The cellular production rate of all prokaryotes on Earth is estimated to be 1.7x10^30^ cells/yr and is highest in the open ocean.

* Do new questions arise from the results?  
What role do these large number of prokaryotes play in maintaining the ecosystem of their various habitats? What is the significance of differences in  turnover rates for prokaryotic populations in different habitats? How can methods to estimate carbon content in prokaryotes be improved upon, especially in habitats where sampling is difficult (eg. deep subsurface)?

* Were there any specific challenges or advantages in understanding the paper (*e.g.* did the authors provide sufficient background information to understand experimental logic, were methods explained adequately, were any specific assumptions made, were conclusions justified based on the evidence, were the figures or tables useful and easy to understand)?  
The paper provided a useful insight regarding the abundance of prokaryotes and their carbon content on Earth. However, the validity of the methods and sources the authors used to estimate these numbers could be more robust in that many interpretations were estimates and extrapolations.


### Problem set 01

[Whitman *et al* 1998](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC33863/)

#### Learning objectives: 
Describe the numerical abundance of microbial life in relation to the ecology and biogeochemistry of Earth systems.

#### Specific questions:
* What are the primary prokaryotic habitats on Earth and how do they vary with respect to their capacity to support life? Provide a breakdown of total cell abundance for each primary habitat from the tables provided in the text.  
The primary prokaryotic habitats on Earth are aquatic environments (cell abundance=1.2x10^29^), soils (cell abundance=2.5x10^29^), and oceanic/terrestrial subsurface (cell abundance=3.8x10^30^). 

* What is the estimated prokaryotic cell abundance in the upper 200 m of the ocean and what fraction of this biomass is represented by marine cyanobacterium including Prochlorococcus? What is the significance of this ratio with respect to carbon cycling in the ocean and the atmospheric composition of the Earth?  
The estimated prokaryotic cell abundance in the upper 200m of the ocean is 3.6x10^28^, with a density of 5x10^5^ cells/ml. The cell density of cyanobacteria is 4x10^4^ cells/ml, so the ratio of cyanobacteria over total prokaryotic cells is 8%.

* What is the difference between an autotroph, heterotroph, and a lithotroph based on information provided in the text?  
An autotroph is "self-nourishing", they fix inorganic carbon (CO2). Heterotrophs assimilate organic carbon, and lithotrophs use inorganic substrates.

* Based on information provided in the text and your knowledge of geography what is the deepest habitat capable of supporting prokaryotic life? What is the primary limiting factor at this depth?  
The deepest habitat capable of supporting prokaryotic life is deep subsurface habitats (terrestrial is 4km and marine is ~14.9km (~4km below the ocean floor. Mariana's Trench is 10.9km below sea level)). The temperature is the limiting factor at this depth. The temperature change is roughly 22 degrees Celsius per km.

* Based on information provided in the text your knowledge of geography what is the highest habitat capable of supporting prokaryotic life? What is the primary limiting factor at this height?  
The highest habitat capable of supporting prokaryotic life is 8.8km at Mount Everest, the highest point on Earth. Although prokaryotes do exist in the stratosphere to some extent such as spore-forming microbes, there are no nutrients and moisture, and there is strong ionizing radiation. 

* Based on estimates of prokaryotic habitat limitation, what is the vertical distance of the Earth's biosphere measured in km?  
14.9km below sea level + 8.8km above sea level = 23.7km is the vertical distance of the Earth's bioshpere.

* How was annual cellular production of prokaryotes described in Table 7 column four determined? (Provide an example of the calculation)  
Population x turnover rate/yr = cells/yr  
3.6x10^28^ cells x 365 days / 16 turnovers = 8.2x10^29^ cells/yr

* What is the relationship between carbon content, carbon assimilation efficiency and turnover rates in the upper 200m of the ocean? Why does this vary with depth in the ocean and between terrestrial and marine habitats?  
Carbon efficiency = 20%, and 5-20fg C/cell (10^-30^pg/cell). There are 3.6x10^28^ cells.  
3.6x10^28^ cells x 20x10^-30^ Pg/cell = 0.72Pg C/marine heterotrophs  
0.72x4 = 2.88Pg/yr  
51Pg C/yr x 85% consumed = 43Pg C  
43Pg C/yr / 2.88Pg/yr = 14.9, 1 turnover every 24.5 days.  
*Numbers were pulled from multiple sources of literature, may not be an accurate calculation*
 
* How were the frequency numbers for four simultaneous mutations in shared genes determined for marine heterotrophs and marine autotrophs given an average mutation rate of 4 x 10^-7^ per DNA replication? (Provide an example of the calculation with units. Hint: cell and generation cancel out)  
(4x10^-7^)^4^=2.56x10^-26^ mutations/generation  
3.1x10^28^ cells x (365/16 turnovers/yr) = 8.2x10^29^ cells/yr  
8.2x10^29^ cells/yr x 2.56x10^-26^ mutations/generation = 2.1x10^4^ mutations/yr

* Given the large population size and high mutation rate of prokaryotic cells, what are the implications with respect to genetic diversity and adaptive potential? Are point mutations the only way in which microbial genomes diversify and adapt?  
With the large population size and high mutation rate, prokaryotic cells have a very high degree of genetic diversity and adaptive potential, as traits that were selected for from the diverse pool were proliferated quickly and in high numbers. In addition to point mutations, microbial genomes also diversify and adapt by lateral gene transfer.

* What relationships can be inferred between prokaryotic abundance, diversity, and metabolic potential based on the information provided in the text?  
The abundance and high diversity of prokaryotes coupled to their ability to rapidly proliferate means they have a high metabolic potential.

### Evidence worksheet 02 

[E. G. Nisbet & N. H. Sleep 2001](https://www.ncbi.nlm.nih.gov/pubmed/?term=The+habitat+and+nature+of+early+life+nisbet)  
[James F. Kasting and Janet L. Siefert 2002](https://www.ncbi.nlm.nih.gov/pubmed/12004117)

#### Learning objectives  
Comment on the emergence of microbial life and the evolution of Earth's systems. 

#### General questions  
*Indicate the key events in the evolution of Earth systems at tic marks on the time series.  
*Describe the dominant physical and chemical characteristics of Earth systems at waypoints.

**4.6 Ga**  Solar system formed. The inner planets received water vapour and carbon from meteorite collisions.  
**4.5 Ga**  Moon formation. The Earth's spin and tilt was established.  
**4.4 Ga**  Oldest minerals (zircons) on Earth were found in Western Australia.  
**4.1 Ga**  First evidence of life found in carbon isotopes stored in zircons.  
**4.0 Ga**  Oldest rock (Acasta gneiss) found in Northwestern Canada.  
**3.8 Ga**  Up until now, meteor impacts shaped the seawater chemistry. Emergence of life from Isua belt in southwest Greenland.  
**3.5 Ga**  Rubisco signiture signifies global oygenic photosynthesis and evolution of cyanobacteria.  
**3.2 Ga**  Microfossils in a volcanogenic sulphide deposit in Western Australia suggests life existed on mid-ocean ridges and microbial life was widespread in the mid-Archaean.  
**3.0 Ga**  Abundant evidence of life present in stromatolites.  
**2.7 Ga**  Great oxidation event.  
**2.5 Ga**  Red rock beds found, signify oxidation events.  
**1.7 Ga**  Emergence of eukaryotes.  
**1.1 Ga**  Snowball Earth occured.  
**540 Ma**  Cambrian explosion occured, increased diversity and complexity of organisms.  
**250 Ma**  Permian extinction caused 95% of species to go extinct.  
**65 Ma**  Cretaceous and Paleogene Extinction caused dinosaurs to go extinct.

Describe the dominant physical and chemical characteristics of Earth systems at he following waypoints:  
**Hadean**    
The sun was a third as bright as it was today. CO2 levels were high and the planet was still young and hot, full of molten rock and lava.  
**Archean**    
The atmosphere was methane rich. Emergence of cyanobacteria meant photosynthesize began and the atmosphere started to become oxygenated.  
**Proterozoic**    
Further oxidation events gave rise to complex eukaryotes. Glaciation events happeneed due to decrease in greenhouse effects.  
**Phanerozoic**    
Cambrian explosion created diverse organisms, but extinction events wiped most out.

### Problem Set 2

[Falkowski, P., Fenchel, T., & Delong, E., 2008](https://www.ncbi.nlm.nih.gov/pubmed/12004117)

#### Learning Objectives
Discuss the role of microbial diversity and formation of coupled metabolism in driving global biogeochemical cycles.

#### Specific Questions:

* What are the primary geophysical and biogeochemical processes that create and sustain condiitons for life on Earth? How do abiotic versus biotic processes vary with respect to matter and energy transformation and how are they interconnected?  
Geophysical processes include tectonic and photochemical processes, supplying substrates and removing products, allowing chemical bonds to form and break in cyclical manner. Biogeochemical reactions are based on H, C, N, O, S, and P building blocks, which are driven by microbially catalyzed redox reactions. Abiotic processes are acid-base and biotic processes are driven by redox reactions.

* Why is Earth's redox state considered an emergent property?  
Feebacks between the evolution of microbial metabolic and geochemical processes create the average redox condition of the oceans and atmosphere.

* How do reversible electron transfer reactions give rise to element and nutrient cycles at different ecological scales? What strategies do microbes use to overcome thermodynamic barriers to reversible elctron flow?  
Reversible electron transfer reactions 

* Using information provided in the text, describe how the nitrogen cycle partitions between different redox "niches" and microbial groups. Is there a relationship between the nitrogen cycle and climate change?  
Nitrogen fixation of N2 to NH4 is required in order for organisms to synthesize nucleic acids and proteins. In the presence of O2, nitrogenase is inhibited and NH4 is oxidized to NO2 then NO3. In the absence of O2, either NO2 or NO3 acts as electron acceptor to produce N2. The nitrogen cycle may cause CO2 levels to drop by converting it to organic matter.

* What is the relationship between microbial diversity and metabolic diversity and how does this relate to the discovery of new protein families from microbial community genomes?  
Microbial diversity allows for catalysis of reversible metabolic pathways. For example, nitrogen fixation is catalyzed by the extremely conserved heterodimeric enzyme complex, nitrogenase, which is inhibited by oxygen. A specific group of bacteria is required to oxidize NH4 to nitrate by oxidizing ammonia to nitrite first, before oxidization to nitrate by a different set of nitrifying bacteria.

* On what basis do the authors consider microbes the guardians of metabolism?  
Dispersal of the core planetary gene set by either vertical or horizontal gene transfer to allow a variety of organisms to guard metabolism. Environmental selection on microbial phenotype leads to evolution of boutique genes that protect the metabolic pathway. The metabolic pathway has a strong chance of surviving in at least one taxon. These taxonomic units may go extinct and evolve but the core machines survive.

### Evidence worksheet 03 

[Rockstrom *et al* 2009](https://www.ncbi.nlm.nih.gov/pubmed/19779433)

#### Learning objectives
Evaluate human impacts on the ecology and biogeochemistry of Earth systems. 

#### General questions
* What were the main questions being asked?  
How far can human development go without causing detrimental or catastrophic effects to the environment? "What are the planetary boundaries"" that define a safe living space on Earth? How many of these boundaries have we already overstepped and what should humans do about it?

* What were the primary methodological approaches used?  
The primary methodological approaches include proposing a framework that contains "planetary boundaries", which are thresholds for each of 9 planetary systems characterized by parameters including climate change and biodiversity loss, that represent the safe operating space for humanity. These thresholds are  critical values taken from literature and are compared and assessed against current values. 

* Summarize the main results or findings.  
The 3 planetary systems of climate change (measured by carbon dioxide concentration), rate of biodiversity loss (measured by extinction rate), and nitrogen cycle (amount of N2 removed from atmosphere for human use) were found to have crossed the proposed planetary boundary. Other system boundaries such as ocean acidification and phosphorus cycle are close to being overstepped.

* Do new questions arise from the results?  
Are there other systems and processes other than the 9 proposed that are important to consider when assessing the impacts of human activity? Could some proposed systems be already overstepped if using different criteria to assess critical values? 

* Were there any specific challenges or advantages in understanding the paper (*e.g.* did the authors provide sufficient background information to understand experimental logic, were methods explained adequately, were any specific assumptions made, were conclusions justified based on the evidence, were the figures or tables useful and easy to understand)?  
The paper provided an insightful view of the state of the anthropocene and provided a good table explicitly showing how each system is assessed and which boundaries are crossed already. However, the authors noted that some boundary thresholds are "merely their best guess", and a more robust method to quantify these thresholds may result in different findings.

### Writing Assessment 01

Microbial life can easily live without us; we, however, cannot survive without the global catalysis and environmental transformations it provides. Microbes have been in existence for much longer than humans have, but with industrialization and other human activities, the human footprint on Earth's environment in our short stint is sizeable enough to alter processes that the sheer number of microbes have established for billions of years. It is the abundance of microbes that exist on Earth along with their high degree of genetic diversity that makes them necessary in sustaining life. Additionally, microbes play a major role in the nitrogen biogeochemical cycle, and with nitrogen being one of the most essential requirements for life to occur, survival without microbes would be challenging. Furthermore, the evolution of complex life such as humans would not have occurred had early microbes not orchestrated pivotal events in Earth's history such as oxygenation events.  
Microbes are very abundant on Earth. There are 4-6 x 1030 prokaryotic cells on Earth which together contain 350-550 Pg of carbon, which is 60-100% the cellular carbon found in plants (Whitman et al., 1998). This substantial number of microbes is distributed in major habitats including aquatic habitats, soil, terrestrial subsurface, as well as marine sediments (Whitman et al., 1998). In addition, they have an average cellular production rate of 1.7 x 1030 cells/yr (Whitman et al., 1998). The significance of such a presence of microbes in the world with rapid proliferation is that they have a much higher potential to accumulate mutations and thus to acquire genetic diversity (Whitman et al., 1998). This is important for global catalysis and environmental transformations because without a high degree of genetic diversity, microbes would not be able to adapt to external stimuli and stresses optimally enough to continue powering Earth's biogeochemical cycles efficiently. Some argue that humans are capable of or even obligated to take on an interventionist and managerial role to guide the Earth's ecosystem, but this would require a world of constant intervention in order to fix problems caused by previous interventions (Achenbach, 2012). If humans had developed chemical engineering technologies to command biogeochemical processes at large scale, it would likely require countless iterations to address flaws and optimize processes. It is at the very least challenging to artificially create these biogeochemical processes that microbes have shaped with their highly adaptive potential for billions of years.  
Nitrogen is the 5th most abundant element in the solar system and necessary for nucleic acid and protein synthesis, and thus to sustain life. In the atmosphere, however, it is in its inert N2 form and must be fixated to ammonium by microbes through a redox reaction (Canfield et al., 2010). Eukaryotes such as legumes and termites also fixate nitrogen, but only in symbiotic association with nitrogen-fixing prokaryotes, signifying the foundational role of prokaryotes in fixating nitrogen (Canfield et al., 2010). Human activity in the 20th century such as industrial processes, agriculture, and burning fossil fuels have all contributed to the rate of nitrogen fixation, doubling the rate and accounting for 45% of total fixed nitrogen annually produced (Canfield et al., 2010). If microbes did not fixate nitrogen and denitrify, nitrogen flux due to human activity would result in much more fixation than denitrification. In fact, the use of fertilizers in agriculture results in fertilizer wash out due to crop inefficiencies which then go on to be converted and leached into rivers and lakes as NO3-, creating hypoxic zones around the world (Canfield et al., 2010). Clearly, not only can we not live without microbe mediation of the nitrogen biogeochemical cycle, but we would also inflict pollution.  
Microbial life has been widespread since the mid-Archaean. Had early microbes not induced important checkpoints during Earth's development, complex organisms such as humans would never have arisen (Nisbet and Sleep, 2003). For example, the evolution of marine cyanobacteria predates 2.7 Ga and is evidence of global oxygenic photosynthesis (Nisbet and Sleep, 2003). In fact, a sharp oxygen spike at 2.2 Ga possibly supported the development of complex eukaryotes.  
The abundance of microbes and their high degree of genetic diversity that makes them necessary in sustaining life. They also play a major role in the nitrogen biogeochemical cycle in sustaining life. Also, the evolution of complex life would not have occurred had early microbes not shaped early Earth. With the intelligence of mankind comes consequences to our environment. Having only been in existence for a minute fraction of time compared to the age of the Earth, humans have shown they are capable of impacting the global ecosystem in yet an even shorter amount of time. Over the past few centuries, the actions of humans such as industrialization and agriculture have influenced if not resulted in climate change, biodiversity loss, and interference with the nitrogen cycle, all of which have put the Earth at risk for detrimentally irreversible change (Rockstrom et al., 2009). Microbes may not be greatly affected by these changes due to their adaptability, but humans will certainly be affected. We must recognize that microbes are essential in maintaining a viable ecosystem and be wary of the environmental impacts of our actions. 


### Module 01 references

Achenbach, J. (2012). Spaceship Earth: A new view of environmentalism. The Washington Post.  
Canfield, D., Glazer, A., & Falkowski, P. (2010). The Evolution and Future of Earth's Nitrogen Cycle. Science, 330(6001), 192-196.  
Falkowski, P., Fenchel, T., & Delong, E. (2008). The Microbial Engines That Drive Earth's Biogeochemical Cycles. Science, 320(5879), 1034-1039.  
Kasting, J. (2003). Life and the Evolution of Earth's Atmosphere. Science, 296(5570), 1066-1068.  
Nisbet, E., & Sleep, N. (2001). The habitat and nature of early life. Nature, 409(6823), 1083-1091.  
Rockström, J., Steffen, W., Noone, K., Persson, Å., Chapin, F., & Lambin, E. et al. (2009). A safe operating space for humanity. Nature, 461(7263), 472-475.
Whitman WB, Coleman DC, and Wiebe WJ. 1998. Prokaryotes: The unseen majority. Proc Natl Acad Sci USA. 95(12):6578-6583.  [PMC33863](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC33863/)



#Module 2

## Module 2 portfolio check
The first of your second level headers (##) is to be used for the portfolio content checks. The Module 01 portfolio check has been built for you directly into this template, but will also be available as a stand-alone markdown document available on the MICB425 GitHub so that you know what is required in each module section in your portfolio. The completion status and comments will be filled in by the instructors during portfolio checks when your current portfolios are pulled from GitHub.

* Problem Set 3
    + Completion status:
    + Comments:
* Portfolio repo setup
    + Completion status: 
    + Comments:
* RMarkdown Pretty PDF Challenge
    + Completion status:
    + Comments:
    
## Data Science Friday

### Problem set_03 - Metagenomics: Genomic Analysis of Microbial Communities
[Wooley *et al* 2010](https://www.ncbi.nlm.nih.gov/pubmed/?term=A+Primer+on+Metagenomics+wooley)

#### Learning objectives: 
Specific emphasis should be placed on the process used to find the answer. Be as comprehensive as possible e.g. provide URLs for web sources, literature citations, etc.  
*(Reminders for how to format links, etc in RMarkdown are in the RMarkdown Cheat Sheets)*

#### Specific Questions:
* How many prokaryotic divisions have been described and how many have no cultured representatives (microbial dark matter)?  


* How many metagenome sequencing projects are currently available in the public domain and what types of environments are they sourced from?
  

* What types of on-line resources are available for warehousing and/or analyzing environmental sequence information (provide names, URLS and applications)?  

* What is the difference between phylogenetic and functional gene anchors and how can they be used in metagenome analysis?   

* What is metagenomic sequence binning? What types of algorithmic approaches are used to produce sequence bins? What are some risks and opportunities associated with using sequence bins for metabolic reconstruction of uncultivated microorganisms?  

* Is there an alternative to metagenomic shotgun sequencing that can be used to access the metabolic potential of uncultivated microorganisms? What are some risks and opportunities associated with this alternative?  


### Evidence Worksheet 04
[Martinez *et al* 2007](https://www.ncbi.nlm.nih.gov/pubmed/17372221)

#### Learning objectives
* Discuss the relationship between microbial community structure and metabolic diversity  
*	Evaluate common methods for studying the diversity of microbial communities.  
*	Recognize basic design elements in metagenomic workflows. 

#### General questions
* What were the main questions being asked?  


* What were the primary methodological approaches used?  


* Summarize the main results or findings.  


* Do new questions arise from the results?  


* Were there any specific challenges or advantages in understanding the paper (*e.g.* did the authors provide sufficient background information to understand experimental logic, were methods explained adequately, were any specific assumptions made, were conclusions justified based on the evidence, were the figures or tables useful and easy to understand)?  


### Evidence Worksheet 05
[Welch *et al* 2002](https://www.ncbi.nlm.nih.gov/pubmed/12471157)

#### Learning objectives
* Evaluate the concept of microbial species based on environmental surveys and
cultivation studies.  
* Explain the relationship between microdiversity, genomic diversity and metabolic potential.  
* Comment on the forces mediating divergence and cohesion in natural microbial
communities. 

#### General questions
* What were the main questions being asked?  
How do the 2 pathogenic and 1 non-pathogenic E. coli strains differ in their pathogenecity of intestinal diseases? From what lineages did pathogenic E. coli arise from? How does lateral gene transfer lead to new uropathogenic E. coli strains? What is the core gene set as opposed to the accessory gene set in each strain?

* What were the primary methodological approaches used?  
Whole genome libraries prepared from genomic DNA and sequenced by Sanger sequencing. Genome sequence was annotated using MAGPIE and ORFs defined using GLIMMER. Predicted proteins were searched with BLAST. Shotgun library sequencing and PCR of CFT073 genome were performed. 

* Summarize the main results or findings.  
Assembly and alignment of CFT073 DNA fragment library to MG1655 and EDL933 showed >70% ORFs previously identified as unique to MG1655 and EDL933 are replaced with CFT073 specific genes. Acquisition of islands resulted in ability to infect urinary tracts and bloodstream, while colonizing the intestine and evading host defenses. Acquired genes promote colonization of specific regions of the intestine and new modes of interaction with host tissue, producing clincally distict variations of gastrointestinal disease.

* Do new questions arise from the results?  
Do the strain specific genes allow its mechanism of infection to differ from other strains? What are the factors (eg. environmental) that influence the rate of gene acquisition? What are some other implications of island acquisition other than ability to infect bloodstream and evade host defenses?

* Were there any specific challenges or advantages in understanding the paper (*e.g.* did the authors provide sufficient background information to understand experimental logic, were methods explained adequately, were any specific assumptions made, were conclusions justified based on the evidence, were the figures or tables useful and easy to understand)?  
Understanding this paper require several read throughs as it was heavily genetics based research. Background info and methods were explained adequately, however more informative figures would offer a better visual interpretation of the results.

### Module 2 References

Martinez, A., Bradley, A., Waldbauer, J., Summons, R., & DeLong, E. (2007). Proteorhodopsin photosystem gene expression enables photophosphorylation in a heterologous host. Proceedings Of The National Academy Of Sciences, 104(13), 5590-5595.  
Welch, R., Burland, V., Plunkett, G., Redford, P., Roesch, P., & Rasko, D. et al. (2002). Extensive mosaic structure revealed by the complete genome sequence of uropathogenic Escherichia coli. Proceedings Of The National Academy Of Sciences, 99(26), 17020-17024.  
Wooley, J., Godzik, A., & Friedberg, I. (2010). A Primer on Metagenomics. Plos Computational Biology, 6(2), e1000667.

## Module 3

### Problem set_04

#### Learning objectives: 

Gain experience estimating diversity within a hypothetical microbial community

#### Outline:

In class Day 1:

1. Define and describe species within your group's "microbial" community.
2. Count and record individuals within your defined species groups.
3. Remix all species together to reform the original community.
4. Each person in your group takes a random sample of the community (*i.e.* devide up the candy).

Assignment:

5. Individually, complete a collection curve for your sample.
6. Calculate alpha-diversity based on your original total community and your individual sample.

In class Day 2:

7. Compare diversity between groups.

#### Part 1: Description and enumeration
Obtain a collection of "microbial" cells from "seawater". The cells were concentrated from different depth intervals by a marine microbiologist travelling along the Line-P transect in the northeast subarctic Pacific Ocean off the coast of Vancouver Island British Columbia. 

Sort out and identify different microbial "species" based on shared properties or traits. Record your data in this Rmarkdown using the example data as a guide.

Once you have defined your binning criteria, separate the cells using the sampling bags provided. These operational taxonomic units (OTUs) will be considered separate "species". This problem set is based on content available at [What is Biodiversity](http://cnx.org/content/m12179/latest/).

```{r message=FALSE}
#To make tables
install.packages("kableExtra")
library(kableExtra)
library(knitr)
#To manipulate and plot data
library(tidyverse)
```

Load in data.

```{r}
example_data1 = data.frame(
  number = c(1,2,3,4,5,6,7,8,9,10,11,12,13),
  name = c("Twizzler", "OG Gummy", "Wine Gummy", "Sugar Gummy", "Sugar Swirls", "Sugar Bottles", "Octopi", "Mike-Ike", "Sphere Candy", "SKittles", "M&Ms", "Kisses", "Lego"),
  characteristics = c("red string", "gummy bear", "wine gummy bear", "sugar gummy bear", "sugar coated swirls", "sugar coated bottle", "sugar coated octopi", "ovoid chewy", "spherical chewy", "flat hardshell chocolate", "bulky hardshell chocolate", "flat-bottom teardrop chocolate", "lego-like hard chewy"), 
  occurences = c(7,16,1,2,1,1,4,25,6,26,41,1,1)
)
```

	
For your community:

* Construct a table listing each species, its distinguishing characteristics, the name you have given it, and the number of occurrences of the species in the collection.
```{r}
example_data1 %>% 
  kable("html") %>%
  kable_styling(bootstrap_options = "striped", font_size = 10, full_width = F)
```


* Ask yourself if your collection of microbial cells from seawater represents the actual diversity of microorganisms inhabiting waters along the Line-P transect. Were the majority of different species sampled or were many missed? 
 
#### Part 2: Collector's curve
To help answer the questions raised in Part 1, you will conduct a simple but informative analysis that is a standard practice in biodiversity surveys. This analysis involves constructing a collector's curve that plots the cumulative number of species observed along the y-axis and the cumulative number of individuals classified along the x-axis. This curve is an increasing function with a slope that will decrease as more individuals are classified and as fewer species remain to be identified. If sampling stops while the curve is still rapidly increasing then this indicates that sampling is incomplete and many species remain undetected. Alternatively, if the slope of the curve reaches zero (flattens out), sampling is likely more than adequate.

To construct the curve for your samples, choose a cell within the collection at random. This will be your first data point, such that X = 1 and Y = 1. Next, move consistently in any direction to a new cell and record whether it is different from the first. In this step X = 2, but Y may remain 1 or change to 2 if the individual represents a new species. Repeat this process until you have proceeded through all cells in your collection.

For your sample:
* Create a collector's curve for your sample (not the entire original community).
```{r}
example_data2 = data.frame(
  x = c(1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19,20,21,22,23,24,25,26,27,28,29,30,31,31,33,34,35,36,37,38,39,40,41,42,43,44,45,46,47,48,49,50,51,52,53,54,55,56,57,58,59,60,61,62,63,64,65,66,67,68,69,70,71,72,73,74,75,76,77,78,79,80,81,82,83,84,85,86,87,88,89,90,91,92,93,94,95,96,97,98,99,100,101,102,103,104,105,106,107,108,109,110,111,112,113,114,115,116,117,118,119,120,121,122,123,124,125,126,127,128,129,130,131),
  y = c(1,1,1,1,1,1,1,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,3,4,4,5,6,7,7,7,7,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,9,9,9,9,9,9,10,10,10,10,10,10,10,10,10,10,10,10,10,10,10,10,10,10,10,10,10,10,10,10,10,10,11,11,11,11,11,11,11,11,11,11,11,11,11,11,11,11,11,11,11,11,11,11,11,11,11,11,11,11,11,11,11,11,11,11,11,11,11,11,11,11,12,13)
)
```

```{r}
library(vegan)
data(example_data2)
rarecurve(example_data2, step = 2,
          sample = min(rowSums(example_data2)),
          col = "blue")
```

* Does the curve flatten out? If so, after how many individual cells have been collected?

The curve flattens out at about 125 individual cells.

* What can you conclude from the shape of your collector's curve as to your depth of sampling?

#### Part 3: Diversity estimates (alpha diversity)
Using the table from Part 1, calculate species diversity using the following indices or metrics.

##### Diversity: Simpson Reciprocal Index

$\frac{1}{D}$ where $D = \sum p_i^2$

$p_i$ = the fractional abundance of the $i^{th}$ species

Own Sample: 
```{r}
species1 = 7/(7+16+1+2+1+1+4+25+6+26+41+1+1)
species2 = 16/(132)
species3 = 1/(132)
species4 = 2/(132)
species5 = 1/(132)
species6 = 1/(132)
species7 = 4/(132)
species8 = 25/(132)
species9 = 6/(132)
species10 = 26/(132)
species11 = 41/(132)
species12 = 1/(132)
species13 = 1/(132)

1 / (species1^2 + species2^2 + species3^2 + species4^2 + species5^2 + species6^2 + species7^2 + species8^2 + species9^2 + species10^2 + species11^2 + species12^2 + species13^2)
```

Original Total Community: 

```{r}
OGcommunity_data1 = data.frame(
  number = c(1,2,3,4,5,6,7,8,9,10,11,12,13),
  name = c("Strings", "OG Gummy", "Wine Gummy", "Sugar Gummy", "Sugar Swirls", "Sugar Bottles", "Octopi", "Mike-Ike", "Sphere Candy", "SKittles", "M&Ms", "Kisses", "Lego"),
  characteristics = c("red string", "gummy bear", "wine gummy bear", "sugar gummy bear", "sugar coated swirls", "sugar coated bottle", "sugar coated octopi", "ovoid chewy", "spherical chewy", "flat hardshell chocolate", "bulky hardshell chocolate", "flat-bottom teardrop chocolate", "lego-like hard chewy"), 
  occurences = c(14,101,8,3,3,3,6,131,19,197,242,16,18)
)
```

```{r}
cspecies1 = 14/(761)
cspecies2 = 101/(761)
cspecies3 = 8/(761)
cspecies4 = 3/(761)
cspecies5 = 3/(761)
cspecies6 = 3/(761)
cspecies7 = 6/(761)
cspecies8 = 131/(761)
cspecies9 = 19/(761)
cspecies10 = 197/(761)
cspecies11 = 214/(761)
cspecies12 = 16/(761)
cspecies13 = 18/(761)

1 / (cspecies1^2 + cspecies2^2 + cspecies3^2 + cspecies4^2 + cspecies5^2 + cspecies6^2 + cspecies7^2 + cspecies8^2 + cspecies9^2 + cspecies10^2 + cspecies11^2 + cspecies12^2 + cspecies13^2)
```
The higher the value is, the greater the diversity. The maximum value is the number of species in the sample, which occurs when all species contain an equal number of individuals. Because the index reflects the number of species present (richness) and the relative proportions of each species with a community (evenness), this metric is a diveristy metric. Consider that a community can have the same number of species (equal richness) but manifest a skewed distribution in the proportion of each species (unequal evenness), which would result in different diveristy values.

* What is the Simpson Reciprocal Index for your sample?
[1] 5.204301
* What is the Simpson Reciprocal Index for your original total community?
[1] 5.114509 


##### Richness: Chao1 richness estimator
Another way to calculate diversity is to estimate the number of species that are present in a sample based on the empirical data to give an upper boundary of the richness of a sample. Here, we use the Chao1 richness estimator. 

$S_{chao1} = S_{obs} + \frac{a^2}{2b})$

$S_{obs}$ = total number of species observed
a = species observed once
b = species observed twice or more

So for our previous example community of 3 species with 2, 4, and 1 individuals each, $S_{chao1}$ =

Own Sample
```{r}
13 + 5^2/(2*8)
```

Original Total Community
```{r}
13 + 0^2/(2*13)
```

* What is the chao1 estimate for your sample?
[1] 14.5625
* What is the chao1 estimate for your original total community?
[1] 13

#### Part 4: Alpha-diversity functions in R
We've been doing the above calculations by hand, which is a very good exercise to aid in understanding the math behind these estimates. Not surprisingly, these same calculations can be done with R functions. Since we just have a species table, we will use the `vegan` package. You will need to install this package if you have not done so previously. 

```{r message=FALSE}
library(vegan)
```

First, we must remove the unnecesary data columns and transpose the data so that `vegan` reads it as a species table with species as columns and rows as samples (of which you only have 1).

```{r}
example_data1_diversity = 
  example_data1 %>% 
  select(name, occurences) %>% 
  spread(name, occurences)

example_data1_diversity
```

Then we can calculate the Simpson Reciprocal Index using the `diversity` function.
```{r}
diversity(example_data1_diversity, index="invsimpson")
```

And we can calculate the Chao1 richness estimator (and others by default) with the the `specpool` function for extrapolated species richness. This function rounds to the nearest whole number so the value will be slightly different that what you've calculated above.
```{r}
specpool(example_data1_diversity)
```

In Project 1, you will also see functions for calculating alpha-diversity in the `phyloseq` package since we will be working with data in that form.

Original Total Community

```{r}
OGcommunity_data1_diversity = 
  OGcommunity_data1 %>% 
  select(name, occurences) %>% 
  spread(name, occurences)

OGcommunity_data1_diversity
```

```{r}
diversity(OGcommunity_data1_diversity, index="invsimpson")
```

```{r}
specpool(OGcommunity_data1_diversity)
```

For your sample:
* What are the Simpson Reciprocal Indices for your sample and community using the R function?

Sample: [1] 5.204301
Community: [1] 4.596235

* What are the chao1 estimates for your sample and community using the R function?
    + Verify that these values match your previous calculations.
Sample:
    Species chao chao.se jack1 jack1.se jack2 boot boot.se n
All      13   13       0    13        0    13   13       0 1

Community:
    Species chao chao.se jack1 jack1.se jack2 boot boot.se n
All      13   13       0    13        0    13   13       0 1

#### Part 5: Concluding activity
If you are stuck on some of these final questions, reading the [Kunin et al. 2010](https://www.ncbi.nlm.nih.gov/pubmed/19725865) and [Lundin et al. 2012](https://www.ncbi.nlm.nih.gov/pubmed/23760801) papers may provide helpful insights. 

* How does the measure of diversity depend on the definition of species in your samples?

* Can you think of alternative ways to cluster or bin your data that might change the observed number of species?


* How might different sequencing technologies influence observed diversity in a sample?



**are the values the same why?
**are the values different why?

Rules for analyzing microbial species:
- use the most updated/curated database 
- filtering out errors, strigent data control
- universal gene amplification (16S rRNA)
  - different regions of amplifcation, make sure to use same (primer bias)
- consistent with DNA preparation for sequencing, use controls  
- consistent sampling methods
- consistent with species definition (i.e. 97% Seq, 100% ASV) 
